numerrin_wrapper/numerrin_wrapper.py

```diff
@@ -0,0 +1,204 @@
+
+from .numerrin_pool import NumerrinPool
+from .numerrin_code import NumerrinCode
+from .numerrin_mesh import NumerrinMesh
+from .numerrin_templates import numname, liccode
+
+import numerrin
+
+
+class NumerrinWrapper(ABCModelingEngine):
+    """ Wrapper to Numerrin
+
+    """
+
+    def __init__(self):
+        super(NumerrinWrapper, self).__init__()
+        numerrin.initlocal("", "PYNUMERRIN_LICENSE", liccode)
+        self.pool = NumerrinPool()
+        self.code = NumerrinCode(self.pool.ph)
+        self._meshes = {}
+        self.CM = DataContainer()
+        self.SP = DataContainer()
+        self.BC = DataContainer()
+        self.CM_extensions = {}
+        self._first = True
+
+    def run(self):
+        """Run Numerrin based on CM, BC and SP data
+
+        Returns
+        -------
+        lastIterationStep : int
+            Last iteration step step taken.
+
+        Raises
+        ------
+        Exception when solver not supported.
+
+        """
+
+        # put SP parameters to pool
+        for key in self.SP:
+            self.pool.put_variable(numname[key], self.SP[key])
+        # parse solver code
+        if self._first:
+            self.code.parse_string(
+                self.code.generate_code(self.CM,
+                                        self.SP,
+                                        self.BC,
+                                        self.CM_extensions))
+            self._first = False
+
+        # execute code
+        self.code.execute(1)
+
+        return self.pool.get_variable('iteration')
+
+    def add_mesh(self, mesh):
+        """Add a mesh to the Numerrin modeling engine.
+
+        Parameters
+        ----------
+        mesh : ABCMesh
+            mesh to be added.
+
+        Returns
+        -------
+        proxy : NumerrinMesh
+            A proxy mesh to be used to update/query the internal representation
+            stored inside the modeling-engine. See get_mesh for more
+            information.
+
+        Raises
+        ------
+        Exception if mesh already exists
+
+        """
+
+        if mesh.name in self._meshes:
+            raise ValueError('Mesh \'{}\` already exists'.format(mesh.name))
+        else:
+            self._meshes[mesh.name] = NumerrinMesh(mesh.name, mesh, self.pool)
+            return self._meshes[mesh.name]
+
+    def get_mesh(self, name):
+        """Get a mesh.
+
+        The returned mesh can be used to query and update the state of the
+        mesh inside the Numerrin modeling engine.
+
+        Parameters
+        ----------
+        name : str
+            name of the mesh to be retrieved.
+
+        Returns
+        -------
+        NumerrinMesh
+
+        Raises
+        ------
+        Exception if mesh not found
+
+        """
+
+        if name in self._meshes:
+            return self._meshes[name]
+        else:
+            raise ValueError(
+                'Mesh \'{}\` does not exist'.format(name))
```

Comment:
Lines 114-118 can be replaced simply with `return self._meshes[name]`
(the dictionary will raise an error if the `name` key does not exist)